### ISSUE TITLE
Interval(-oo, oo) is not S.Reals

### ISSUE THREAD

**Issue by smichr** [sympy/sympy#10143]
It seems like the Interval created from S.Reals should be S.Reals

```python
>>> Interval(*S.Reals.args) is S.Reals
False
>>> Interval(*S.Reals.args) == S.Reals
True
```

**Comment by asmeurer** [comment]
I agree that it's weird, since S.Reals is an Interval, but it seems that it would be useful to be able to represent the two differently, at least for printing. That probably means that S.Reals shouldn't really be an Interval, though. This does break the pattern for other singleton objects.

**Comment by jksuom** [comment]
Here are some additional experiments.
```
>>> S.Reals.args
(-oo, oo, True, True)
>>> Interval(-oo, oo).args
(-oo, oo, True, True)
>>> Interval(*Interval(-oo, oo).args) is Interval(-oo, oo)
False
>>> Interval(*Interval(-oo, oo).args) == Interval(-oo, oo)
True
```

**Comment by Curious72** [comment]
Performed the following experiment ...
This computes to true
>>> t=S.Reals
>>> S.Reals is t
True
>>> t
(-oo, oo)
>>>

**Comment by Curious72** [comment]
I guess u are comparing the objects of the wrong type , that is why it is giving an error
Interval(-oo, oo) is of type sympy.set.sets.Interval while S.Reals is of type sympy.sets.fancysets.Reals,
so comparing objects of two different types is returning false, i guess thats the problem

**Comment by darkcoderrises** [comment]
I added a print statement to as to when a new S.Reals class object is created. I noticed that 
In [19]: Interval(*S.Reals.args) is S.Reals
Out[19]: False

In [20]: Interval(*S.Reals.args) == S.Reals
other
Out[20]: True

Interval creates a new object, so it comes true only when new S.Reals object is created.

**Comment by kshitij10496** [comment]
```
[In]: Interval(*S.Reals.args) is S.Reals
[Out]: False
```
This is because the `id` values of both these objects are different.

```
[In] : Interval(*S.Reals.args) == S.Reals
[Out] : True
```
This is because the both objects hash to the same number as they have the same values i.e. `(-oo, oo)`

Moreover, I think since the types of `Interval(*S.Reals.args)` and `S.Reals` are different, their `id`s are correctly different.   
Hence, `is` returns `False`.
